linguatec_lexicon/models.py

```diff
@@ -46,28 +46,44 @@ def _clean_search_query(self, query):
 
         return query
 
-    def search(self, query):
+    def search(self, query, lex=None):
         MIN_SIMILARITY = 0.3
         query = self._clean_search_query(query)
 
+        #Get and use the key of the Lexicon instead of the name
+        if lex is not None:
+            key_lex = Lexicon.objects.get(name=lex)
+
         if connection.vendor == 'postgresql':
             iregex = r"\y{0}\y"
         elif connection.vendor == 'sqlite':
             iregex=r"\b{0}\b"
-            return self.filter(term__iregex=iregex.format(query))
+            if lex is None:
+                return self.filter(term__iregex=iregex.format(query))
+            else:
+                return self.filter(term__iregex=iregex.format(query),lexicon=key_lex)
         else:
             filter_query = (
                 Q(term=query) |
                 Q(term__startswith=query) |
                 Q(term__endswith=query)
             )
-            return self.filter(filter_query)
+            if lex is None:
+                return self.filter(filter_query)
+            else:
+                return self.filter(filter_query,lexicon=key_lex)
 
         # sort results by trigram similarity
-        qs = self.filter(
-                term__iregex=iregex.format(query)
-            ).annotate(similarity=TrigramSimilarity('term', query)
-            ).filter(similarity__gt=MIN_SIMILARITY).order_by('-similarity')
+        if lex is None:
+            qs = self.filter(
+                    term__iregex=iregex.format(query)
+                ).annotate(similarity=TrigramSimilarity('term', query)
+                ).filter(similarity__gt=MIN_SIMILARITY).order_by('-similarity')
+        else:
+            qs = self.filter(
+                    term__iregex=iregex.format(query),lexicon=key_lex
+                ).annotate(similarity=TrigramSimilarity('term', query)
+                ).filter(similarity__gt=MIN_SIMILARITY).order_by('-similarity')
```

Comment:
Keep in mind that a [QuerySet is evaluated in a lazy way](https://docs.djangoproject.com/en/3.1/ref/models/querysets/#when-querysets-are-evaluated), so it could be built step by step without impact performance.

So, I guess that all this changes could be summarized with:
```python
# line 54
        if lex is not None:
            lexicon = Lexicon.objects.get(name=lex)
            qs = self.filter(lexicon=lexicon)
        else:
            qs = self

# other lines, revert changes to original code
```

Reply:
Yeah, it is definitely a better way to code the search method. Change made.